docs/rfcs/0007-deployment-chain.md

```diff
@@ -0,0 +1,275 @@
+- Start Date: 2021-11-03
+- Target Version: 0.22.0
+
+# Summary
+This RFC introduces a new way to enable users to connect multiple applications to a complex deployment pipeline. It can bring to users the ability to roll out to multiple clusters gradually or promote across environments.
+
+# Motivation
+With the current supporting deployment, our users have to define their application as one of the supporting application's kinds (K8s, Terraform, CloudRun, Lambda, ECS), they can still define multiple PipeCD applications as parts of their application and trigger them one by one, but it's a bit difficult for them to control the deployment chain smoothly.
+
+![image](assets/deployment-chain-kind.png)
+Use-case <1>: chain of deployments contains multiple application cross kinds
+
+Another typical use-case is when users want to build up some kind of chain such as deploy application to the development environment, then if it's done successfully, deploy that application to the staging and then the production environment.
+
+![image](assets/deployment-chain-env.png)
+Use-case <2>: chain of deployments contains application deployed across multiple environments
+
+Or a variant of that requirement, when users want to roll out their applications to their cluster one by one based on its region. As the same below image, the order of the rolling out deployment should be flex, supports both sequential and parallel.
+
+![image](assets/deployment-chain-region.png)
```

Comment:
All the application names in this image should be fixed as `application-x-region-a`, `application-x-region-b`, ...

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2743/commits/b16dee1326f8f0409ed144aa8979c515db8a9061